Tools/SwiftBrowser/Source/ContentView.swift

```diff
@@ -55,6 +55,67 @@ fileprivate struct ToolbarBackForwardMenuView: View {
     }
 }
 
+fileprivate struct DialogActionsView: View {
+    private let dialog: DialogPresenter.Dialog
+
+    @State private var promptText = ""
+
+    init(dialog: DialogPresenter.Dialog) {
+        self.dialog = dialog
+
+        if case let .prompt(_, defaultText, _) = dialog.configuration, let defaultText {
+            _promptText = State(initialValue: defaultText)
+        }
+    }
+
+    var body: some View {
+        switch dialog.configuration {
+        case let .alert(_, dismissAlert):
+            Button("Dismiss") {
+                dismissAlert()
+            }
+
+        case let .confirm(_, dismissConfirm):
+            Button("OK") {
+                dismissConfirm(.ok)
+            }
+
+            Button("Cancel", role: .cancel) {
+                dismissConfirm(.cancel)
+            }
+
+        case let .prompt(_, _, dismissPrompt):
+            TextField("Text", text: $promptText)
+
+            Button("OK") {
+                dismissPrompt(.ok(promptText))
+            }
+            .keyboardShortcut(.defaultAction)
+
+            Button("Cancel", role: .cancel) {
+                dismissPrompt(.cancel)
+            }
+        }
+    }
+}
+
+fileprivate struct DialogMessageView: View {
+    let dialog: DialogPresenter.Dialog
+
+    var body: some View {
+        switch dialog.configuration {
+        case let .alert(message, _):
```

Comment:
This is just sample code, so N.B.D., but another way to handle this would be making a `var message: String { … }` extension on the enum so that you could write `Text(dialog.configuration.message)` here. That's mostly isomorphic, but it does simplify your view tree. That only applies because the types here are all otherwise identical, so if there were any other customizations I probably wouldn't recommend it. Just throwing it out.